src/query/service/src/pipelines/processors/transforms/hash_join/transform_hash_join_probe.rs

```diff
@@ -121,6 +124,70 @@ impl TransformHashJoinProbe {
         Ok(())
     }
 
+    fn final_merge_into_partial_unmodified_scan(
+        &mut self,
+        item: ChunkPartialUnmodified,
+    ) -> Result<()> {
+        // matched whole block, need to delete
+        if item.0.is_empty() {
+            let prefix = item.1;
+            let (segment_idx, block_idx) = split_prefix(prefix);
+            info!(
+                "matched whole block: segment_idx: {}, block_idx: {}",
+                segment_idx, block_idx
+            );
+            let data_block = DataBlock::empty_with_meta(Box::new(BlockMetaIndex {
+                segment_idx: segment_idx as usize,
+                block_idx: block_idx as usize,
+                inner: None,
+            }));
+            self.output_data_blocks.push_back(data_block);
+            return Ok(());
+        }
+
+        let chunks_offsets = unsafe {
+            &*self
+                .join_probe_state
+                .hash_join_state
+                .merge_into_state
+                .chunk_offsets
+                .get()
+        };
+        let build_state = unsafe { &*self.join_probe_state.hash_join_state.build_state.get() };
+        let chunk_block = &build_state.generation_state.chunks[item.1 as usize];
+        let chunk_start = if item.1 == 0 {
+            0
+        } else {
+            chunks_offsets[(item.1 - 1) as usize]
+        };
+        for (interval, prefix) in item.0 {
+            let indices = ((interval.0 - chunk_start)..=(interval.1 - chunk_start))
```

Comment:
the collect could be overload, why not using iter directly?

```
for start in ((interval.0 - chunk_start)..= (interval.1 - chunk_start)).step_by(self.max_block_size) {
   let end = (interval.1 - chunk_start).min(start + self.max_block_size);
   ...
}
```

Reply:
well, this is really a bad way, let me change this.